tests/test_openml/test_config.py

```diff
@@ -29,15 +29,29 @@ def test_non_writable_home(self, log_handler_mock, warnings_mock):
         assert not log_handler_mock.call_args_list[0][1]["create_file_handler"]
         assert openml.config._root_cache_directory == Path(td) / "something-else"
 
-    @unittest.mock.patch("os.path.expanduser")
-    def test_XDG_directories_do_not_exist(self, expanduser_mock):
+    def test_XDG_directories_do_not_exist(self):
         with tempfile.TemporaryDirectory(dir=self.workdir) as td:
+            # Save previous state
+            _prev = os.environ.get("XDG_CONFIG_HOME")
```

Comment:
You shouldn't need to save this, right? `os.environ` shouldn't persist to the environment variables.

Reply:
Unfortunately it persists between functions calls. Technically I shouldn't need to save it but just wanted to reset the state to where it was exactly. I would leave this as is

Reply:
Then it should be good to create it as a context manager instead so that it resets it when an error happens in between the save and restore.